src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/HelperMatcher.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Razor
+{
+    /// <summary>
+    /// Determines whether a Razor document contains @helper functions
+    /// and produces local method replacements for the @helpers.
+    /// </summary>
+    public class HelperMatcher : IHelperMatcher
+    {
+        // Regex used to identify @helper functions
+        private static readonly Regex HelperDeclRegex = new Regex(@"@helper\s+.*?\(.*?\)", RegexOptions.Compiled);
```

Comment:
Will this match if there are nested parentheses?

Reply:
It actually doesn't need to worry about nested parentheses because we're actually only working with the `@helper (` part of the declaration. I'll change the regex to only match up to there to make that clearer.